### PR TITLE
Updates - github actions

### PR DESCRIPTION
**Pull request by GernotMaier**
Several of the action flows are outdated:

- general update of version names 
- replace conda installation 'by hand' in CI-docs.yml by using mamba-org/setup-micromamba
- replace mamba-org/provision-with-micromamba by mamba-org/setup-micromamba (as indicated [here](https://github.com/marketplace/actions/provision-with-micromamba)

Note that it is interesting that there is no `:latest` flag for github actions and that one has to manually update the versions. There is a github action to update github action ([here](https://github.com/marketplace/actions/github-actions-version-updater)), but let's keep it simple - github is provided a deprecated warning:

Screenshot 2023-08-11 at 15 27 56

### CODE REVIEW

File: .github/workflows/CI-unittests.yml
Comment (VictorBarbosaMartins) [inline .github/workflows/CI-unittests.yml:81]: In case we update the version to 1.4.3, please update it here too

**Review by VictorBarbosaMartins — APPROVED** [review]
Hi @GernotMaier. Thanks for this update. 
I am not the expert on github workflows but I tried my best. I wrote a couple of questions/suggestions, it is up to you to consider them or go on. I believe we will see the effect of this PR, especially on the docs, after we merge it.

I am already approving but let's at least wait for the tests to complete (I merged main into the branch).

File: .github/workflows/CI-docs.yml
Comment (GernotMaier) [reply .github/workflows/CI-docs.yml:20]: Yes, unfortunately we have to. This is why we introduced the bot to check for version on certain time intervals and update it.

File: .github/workflows/CI-docs.yml
Comment (GernotMaier) [reply .github/workflows/CI-docs.yml:26]: Not really, but it has saved me quite a few times while debugging. I think this is is a easy test that the installation went well.

File: .github/workflows/CI-docs.yml
Comment (GernotMaier) [reply .github/workflows/CI-docs.yml:19]: This is for the docs, I think here the python version is not so relevant (unless there is a python / sphinx dependencies). I think at this point we want things working with 3.9-3.11, so this should be goo.d

**Pull request merged**
